robot/client/client.go

```diff
@@ -71,17 +73,105 @@ type RobotClient struct {
 	closeContext context.Context
 }
 
+var exemptFromConnectionCheck = map[string]bool{
+	"/proto.rpc.webrtc.v1.SignalingService/Call":                 true,
+	"/proto.rpc.webrtc.v1.SignalingService/CallUpdate":           true,
+	"/proto.rpc.webrtc.v1.SignalingService/OptionalWebRTCConfig": true,
+	"/proto.rpc.v1.AuthService/Authenticate":                     true,
+}
+
+func skipConnectionCheck(method string) bool {
+	return exemptFromConnectionCheck[method]
+}
+
+func isClosedPipeError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), io.ErrClosedPipe.Error())
```

Comment:
nit: should be able to do `errors.Is` instead of a straight string match

Reply:
sadly we need to do the string match since the pipe closed error we receive is constructed from [status.Error](https://pkg.go.dev/google.golang.org/grpc@v1.50.1/status#Error) method, so it's not actually the same error.

Reply:
so sad